sw/host/opentitanlib/src/transport/cw310/usb.rs

```diff
@@ -608,19 +602,18 @@ impl Backend {
     }
 }
 
-lazy_static! {
-    // Mapping of SAM3 pin names to pin numbers.
-    static ref SAM3X_PIN_NAMES: HashMap<&'static str, u8> = collection! {
-        "PA0" =>  0,
-        "PA1" =>  1,
-        "PA2" =>  2,
-        "PA3" =>  3,
-        "PA4" =>  4,
-        "PA5" =>  5,
-        "PA6" =>  6,
-        "PA7" =>  7,
-        "PA8" =>  8,
-        "PA9" =>  9,
+fn sam3x_pin_from_name(name: &str) -> Option<u8> {
```

Comment:
Can these functions not be `const`?

Reply:
Not yet, traits can't contain `const fn`s, so the `Eq` trait isn't `const` which is required for strings in `match` statements.